### PR TITLE
Deprecate toplevel mpl.text.get_rotation; normalize rotations early.

### PR DESCRIPTION
**Pull request by anntzer**
get_rotation had been made a toplevel function a long time ago to be
used in TextWithDash, which has now been removed, so there isn't much
justification to have it separate.  Also, note that while the toplevel
get_rotation's implementation also supported string-form of floats (see
test_text.test_get_rotation_string), this was not consistent with the
docstring, and, more importantly, it was not possible to set a Text's
rotation to e.g. "15." because Text.set_rotation() would first reject
that anyways.

Also, we can directly normalize angles in Text.set_rotation, rather than
doing it again and again in Text.get_rotation.  Again, this made the old
inconsistency (on supporting string-form floats) clearer.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/tests/test_text.py [hunk 1]
```diff
@@ -345,33 +345,32 @@ def test_non_default_dpi(text):
...
 def test_get_rotation_int():
     for i in [67, 16, 41]:
-        assert mpl.text.get_rotation(i) == float(i)
+        assert Text(rotation=i).get_rotation() == float(i)
 
 
 def test_get_rotation_raises():
     with pytest.raises(ValueError):
-        mpl.text.get_rotation('hozirontal')
+        Text(rotation='hozirontal').get_rotation()
```
Comment (oscargus) [inline lib/matplotlib/tests/test_text.py:364]: Shouldn't this error really trigger when creating the `Text` object? (Not part of this PR, but it looked a bit strange that you actually had to do `get_rotation` to trigger it, but you do.)

**Review by oscargus — APPROVED** [review]

File: lib/matplotlib/tests/test_text.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/tests/test_text.py:364]: Oh, I realize that with this change it will actually be triggered in the object creation. So maybe remove the `get_creation`?

Comment (anntzer) [reply lib/matplotlib/tests/test_text.py:364]: good point, done

**Review by QuLogic — APPROVED** [review]
